app/script.js

```diff
@@ -0,0 +1,76 @@
+// random-string.js
+
+$(document).ready(function() {
+
+  $("#randomButton").on("click", function() {
+    $.get("/strings", function(response) {
+      $("#stringDisplay").text(response);
+    });
+  });
+
+  // Hover Event Listener - Receives JSON
+  $("#externalSiteInfo").on("mouseover", function() {
+    $.get("/npmname", function(response) {
+      $("#externalSiteInfo").text(JSON.parse(response).text);
+    });
+  });
+  $("#externalSiteInfo").on("mouseout", function() {
+    setTimeout( function() {
+      $("#externalSiteInfo").text("Put your mouse over this text to see their undisclosed text.");
+    }, 3000 );
+  });
+
+  /////////// Button Event Listeners /////////////
+  // Random Strings/Jokes
+  $("button").on("click", function() {
+    var urlKey = $(this).attr("id");
+    var resText;
+
+    $.get(urlKey, function(response) {
+      if ( typeof response === "object" ) {
+        resText = response.setup + ":" + response.punchline;
+      } else {
+        resText = response;
+      }
+
+      $("#stringDisplay").text(resText);
+    });
+  });
+
+  $("#piglatin").on("submit", function(e) {  // "e" stands for returned Event
+    e.preventDefault(); // Keeps it from reloading page!
+    var firstname = $("input[name=firstname]").val();
+    var lastname = $("input[name=lastname]").val();
+    var name = { firstname: firstname, lastname: lastname };
+
+    $.post("/piglatin", name, function(response) {
+      // Why build here? Format data IN THE VIEW, NOT IN THE CONTROLLER.
```

Comment:
yes, for the most part. If you need a fair amount of data-wrangling (e.g. you needed to use some JavaScript string methods on the data), then you'd do it in the controller. But then again, it all depends on how the framework you're using conceptualizes  the "view" and the "controller."